src/Profile.svelte

```diff
@@ -0,0 +1,380 @@
+<script lang="ts">
+  import type { SvelteComponent } from 'svelte';
+  import type { Config } from '@app/config';
+  import Address from '@app/Address.svelte';
+  import Avatar from '@app/Avatar.svelte';
+  import Icon from '@app/Icon.svelte';
+  import SetName from '@app/ens/SetName.svelte';
+  import SeedAddress from '@app/SeedAddress.svelte';
+  import TransferOwnership from '@app/base/orgs/TransferOwnership.svelte';
+  import Link from '@app/Link.svelte';
+  import { Profile, ProfileType } from '@app/profile';
+  import Loading from '@app/Loading.svelte';
+  import * as utils from '@app/utils';
+  import { session } from '@app/session';
+  import { Org } from '@app/base/orgs/Org';
+  import Message from '@app/Message.svelte';
+  import Error from '@app/Error.svelte';
+  import { User } from '@app/base/users/User';
+  import Projects from '@app/base/orgs/View/Projects.svelte';
+
+  export let config: Config;
+  export let addressOrName: string;
+  export let action: string | null = null;
+
+  let setNameForm: typeof SvelteComponent | null =
+    action === "setName" ? SetName : null;
+  const setName = () => {
+    setNameForm = SetName;
+  };
+
+  let transferOwnerForm: typeof SvelteComponent | null = null;
+  const transferOwnership = () => {
+    transferOwnerForm = TransferOwnership;
+  };
+  $: account = $session && $session.address;
+  $: isOwner = (org: Org): boolean => $session
+    ? utils.isAddressEqual(org.owner, $session.address)
+    : false;
+  $: getOrgTreasury = async (org: Org): Promise<Array<utils.Token>| undefined> => {
+    const addressType = await utils.identifyAddress(org.owner, config);
+    // We query the org treasury only for Gnosis Safes, to maintain some privacy for EOA org owners.
+    if (addressType === utils.AddressType.Safe) {
+      try {
+        const tokens = await utils.getTokens(org.owner, config);
+        const balance = await config.provider.getBalance(org.owner);
+
+        if (! balance.isZero()) {
+          // To maintain the format we hardcode the ETH specs.
+          return [{ balance, decimals: 18, logo: "", name: "Ethereum", symbol: "ETH" }, ...tokens];
+        } else {
+          return tokens;
+        }
+      } catch (e) {
+        console.error(e);
+      }
+    }
+  };
+  $: isUserAuthorized = (address: string): boolean | null => {
+    return $session && utils.isAddressEqual(address, $session.address);
+  };
+  $: isOrgAuthorized = async (org: Org): Promise<boolean> => {
+    if ($session) {
+      if (isOwner(org)) {
+        return true;
+      }
+      return await org.isMember($session.address, config);
+    }
+    return false;
+  };
```

Comment:
I know this wasn't introduced in this PR, but why is it that all these functions are defined as reactive?

Reply:
They are required to change the permissions on setting profile and changing owner, eg when the logged in user changes.

Reply:
Hmm not sure that answers my question, since they are functions that would have to be called anyway.. I'll look into it.